ngx_rtmp_relay_module.c

```diff
@@ -361,7 +390,21 @@ ngx_rtmp_relay_create_connection(ngx_rtmp_conf_ctx_t *cctx, ngx_str_t* name,
         goto clear;
     }
 
-    if (ngx_rtmp_relay_copy_str(pool, &rctx->url, &target->url.url) != NGX_OK) {
+    url = &target->url;
+
+    if(session && ngx_strlchr(url->url.data, url->url.data + url->url.len, '$')) {
+        ngx_str_t *u = ngx_pcalloc(pool, sizeof(ngx_str_t));
```

Comment:
It's better to place variable definitions at where the function body begins.

Reply:
I mean "ngx_str_t *u"!

Reply:
sorry, my bad^_^